### ISSUE TITLE
get_key_value_changes can panic when processing a query

### ISSUE THREAD

**Issue by bowenwang1996** [nearprotocol/nearcore#2052]
On staging network it panicked at https://github.com/nearprotocol/nearcore/blob/143b138ab153800de227726c5cd079b8ba39cd5e/chain/chain/src/store.rs#L968 with
```
thread 'main' panicked at 'slice index starts at 80 but ends at 48', src/libcore
/slice/mod.rs:2589:5
```

**Comment by nearmax** [comment]
@bowenwang1996  Were we able to reproduce it? How did we identify that it broke at that specific line of code? I saw the slice error but were not able to attribute to that specific line of code.

**Comment by frol** [comment]
Duplicate of #2048 (resolved in #2148).
